lib/ajax-datatables-rails/orm/active_record.rb

```diff
@@ -38,31 +38,28 @@ def composite_search(records)
         records
       end
 
-      def build_conditions_for(query)
+      def build_conditions_for(query, regex)
         search_for = query.split(' ')
         criteria = search_for.inject([]) do |criteria, atom|
-          criteria << searchable_columns.map { |col| search_condition(col, atom) }
+          criteria << searchable_columns.map { |col| search_condition(col, atom, regex == 'true') }
             .reduce(:or)
         end.reduce(:and)
         criteria
       end
 
       def aggregate_query
-        conditions = searchable_columns.each_with_index.map do |column, index|
+        conditions = view_columns.each_with_index.map do |column, index|
```

Comment:
I don't think that's completely accurate... we don't need an index for searching columns, `searchable_columns` will return an array of columns to build our `where` conditions.

So, in your example where columns 0,1,2,4 are searchable and 3 is not (collected from the columns description via the params hash), using the `searchable_columns` method will yield us precisely that: columns 0,1,2,4 and so, we then already have the _actual_ columns to build our where clause(s), so it's safe to only iterate them.

This is, we only need the `ModelName.field_name` from a _pre-filtered_ columns array, based on whether their description says they're `searchable = "true"` or not.

Keeping track of the actual column index is important _only_ for sorting purposes, I think. But is really not needed for filtering records... here we can collapse all `view_columns` into those which really are described as `searchable` by the params hash.

Hope to make sense :)